bin/datasource-invoke.js

```diff
@@ -22,9 +22,15 @@ process.once('message', function(msg) {
     send({
       error: err,
       callbackArgs: args,
-    });
-
-    process.nextTick(function() {
+    }, function() {
+      // XXX better to do this when the send has completed, which may be the
+      // actual source of your bug.
+      //
+      // But even better would be to not do this here. kill your child in your
+      // parent process after its received its message (assuming that you don't
+      // control the invoked methods... they shouldn't have side effects, and
+      // node should exit by itself, but if you don't trust them, best to kill
+      // from the parent where it cannot be stopped).
```

Comment:
That sortof makes sense... except that connection pools should be unrefed, I think, so they don't hold node open if there is no server listening. But that is probably a fantasy world, there will always be bugs, its not a bad idea to kill from parent.